spock-core/src/main/java/org/spockframework/runtime/model/BlockInfo.java

```diff
@@ -27,6 +27,14 @@ public class BlockInfo {
   private BlockKind kind;
   private List<String> texts;
 
+  public BlockInfo() {
+  }
+
+  public BlockInfo(BlockKind kind, List<String> texts) {
+    this.kind = kind;
+    this.texts = texts;
+  }
+
   public BlockKind getKind() {
     return kind;
   }
```

Comment:
The fields (including `List<String>`) could be `null` also before, but making changes in that class, you might also want to add `@Nullable` to the constructor arguments (if `requireNotNull()` will not be used) and to the getters (as there is still the old default constructor).

Reply:
I'm not sure if we can just remove the no-args constructor as well as the setters, maybe there is something in the wild that relies on them.